test/e2e/storage/testsuites/multiattach.go

```diff
@@ -0,0 +1,272 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testsuites
+
+import (
+	"math/rand"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/storage/testpatterns"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+)
+
+type multiAttachTestSuite struct {
+	tsInfo TestSuiteInfo
+}
+
+var _ TestSuite = &multiAttachTestSuite{}
+
+// InitMultiAttachTestSuite returns multiAttachTestSuite that implements TestSuite interface
+func InitMultiAttachTestSuite() TestSuite {
+	return &multiAttachTestSuite{
+		tsInfo: TestSuiteInfo{
+			name: "multiAttach",
+			testPatterns: []testpatterns.TestPattern{
+				testpatterns.FsVolModeDynamicPV,
+				testpatterns.BlockVolModeDynamicPV,
+				// Currently, multiple volumes are not generally available for pre-provisoined volume,
+				// because containerized storage servers, such as iSCSI and rbd, are just returning
```

Comment:
I'm going to make it so the iscsi image creates 2 luns instead of 1. otherwise we cannot test the case where 2 vols are different except for the lun, I want to add that testcase here